src/index.js

```diff
@@ -0,0 +1,311 @@
+const core = require('@actions/core');
+const fs = require('fs');
+const crypto = require('crypto');
+const { SeqeraPlatformAPI } = require('./seqera-api');
+
+/**
+ * Create log file with timestamp matching entrypoint.sh format
+ */
+function createLogFile() {
+  const now = new Date();
+  const timestamp = now.toISOString()
+    .replace(/T/, '-')
+    .replace(/:/g, '_')
+    .replace(/\..*/, '')
+    .replace(/-/g, '_');
+  return `tower_action_${timestamp}.log`;
+}
+
+/**
+ * Create JSON file with UUID matching entrypoint.sh format
+ */
+function createJsonFile() {
+  const uuid = crypto.randomUUID();
+  return `tower_action_${uuid}.json`;
+}
+
+/**
+ * Write message to log file and console
+ */
+function logMessage(logFile, message, level = 'INFO') {
+  const timestamp = new Date().toISOString();
+  const logEntry = `[${timestamp}] ${level}: ${message}\n`;
+  
+  try {
+    fs.appendFileSync(logFile, logEntry);
+  } catch (error) {
+    core.error(`Failed to write to log file: ${error.message}`);
+  }
+}
+
+/**
+ * Main action entry point
+ */
+async function run() {
+  let logFile, jsonFile;
+  
+  try {
+    // Create log files
+    logFile = createLogFile();
+    jsonFile = createJsonFile();
+    
+    logMessage(logFile, 'Starting Seqera Platform workflow launch', 'INFO');
+    core.info('🚀 Starting Seqera Platform workflow launch');
+    
+    // Get action inputs
+    const inputs = {
+      accessToken: core.getInput('access_token', { required: true }),
+      pipeline: core.getInput('pipeline') || `https://github.com/${process.env.GITHUB_REPOSITORY}`,
+      workspaceId: core.getInput('workspace_id'),
+      computeEnv: core.getInput('compute_env'),
+      apiEndpoint: core.getInput('api_endpoint') || 'https://api.cloud.seqera.io',
+      revision: core.getInput('revision'),
+      workdir: core.getInput('workdir'),
+      parameters: core.getInput('parameters'),
+      profiles: core.getInput('profiles'),
+      runName: core.getInput('run_name'),
+      nextflowConfig: core.getInput('nextflow_config'),
+      preRunScript: core.getInput('pre_run_script'),
+      labels: core.getInput('labels'),
+      wait: core.getBooleanInput('wait'),
+      debug: core.getBooleanInput('debug')
+    };
+    
+    // Mask sensitive data
+    core.setSecret(inputs.accessToken);
+    if (inputs.workspaceId) core.setSecret(inputs.workspaceId);
+    
+    // Log input configuration
+    logMessage(logFile, `Pipeline: ${inputs.pipeline}`);
+    logMessage(logFile, `API Endpoint: ${inputs.apiEndpoint}`);
+    logMessage(logFile, `Workspace ID: ${inputs.workspaceId || '<not set>'}`);
+    logMessage(logFile, `Compute Environment: ${inputs.computeEnv || '<not set>'}`);
+    
+    // Debug mode logging
+    if (inputs.debug) {
```

Comment:
I wonder if we should always print these? Otherwise if it fails the end user has to re-run with debug enabled, rather than just checking the log to see what was wrong.

If we're worried about being too verbose, we can group the log messages: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-commands#grouping-log-lines

Then we can just remove the debug option entirely and have it always be verbose, but collapsed / grouped..?